.buildkite/scripts/git_setup.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -ex
+
+export GIT_BRANCH=${BUILDKITE_BRANCH}
+
+git switch -
+git checkout $GIT_BRANCH
+git pull origin $GIT_BRANCH
+git config --local user.email 'elasticmachine@users.noreply.github.com'
+git config --local user.name 'Elastic Machine'
```

Comment:
Overall - looks good, but the git configs for the email and name should already be set up in the pre-command hook here: https://github.com/elastic/thumbnails4j/blob/d9aa68eb529aefda3c74ab7cb8a3fb53900c708b/.buildkite/hooks/pre-command#L49

Reply:
Thread ongoing in slack: https://elastic.slack.com/archives/C01DJ3DC0AX/p1741622913039519?thread_ts=1741360830.306909&cid=C01DJ3DC0AX

Reply:
going to merge this as is, downgrade, release, then follow-up in https://github.com/elastic/search-developer-productivity/issues/3753